### ISSUE TITLE
JSON Parser should not remove keys from log record

### ISSUE THREAD

**Issue by PettitWesley** [fluent/fluent-bit#1605]
**Is your feature request related to a problem? Please describe.**
TL:DR I believe that the [kubernetes filter Merge_Log option](https://docs.fluentbit.io/manual/filter/kubernetes#configuration-parameters) is basically what I want, but outside of Kubernetes.

I'm using Fluent Bit with Docker and [Amazon ECS's new FireLens feature](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html). My container produces logs as serialized JSON. I use a filter with the JSON parser, which works great, but it removes all fields not in the serialized JSON.

Without JSON parsing, the log record sent by Fluent Bit looks like this:

```
{
    "source": "stdout",
    "log": "{\"requestID\": \"b5d716fca19a4252ad90e7b8ec7cc8d2\", \"requestInfo\": {\"ipAddress\": \"204.16.5.19\", \"path\": \"/activate\", \"user\": \"TheDoctor\"}}",
    "container_id": "e54cccfac2b87417f71877907f67879068420042828067ae0867e60a63529d35",
    "container_name": "/ecs-demo-6-container2-a4eafbb3d4c7f1e16e00"
    "ecs_cluster": "mycluster",
    "ecs_task_arn": "arn:aws:ecs:us-east-2:01234567891011:task/mycluster/3de392df-6bfa-470b-97ed-aa6f482cd7a6",
    "ecs_task_definition": "demo:7"
    "ec2_instance_id": "i-06bc83dbc2ac2fdf8"
}
```

But with JSON Parsing, I get:

```
{
  "requestID": "b5d716fca19a4252ad90e7b8ec7cc8d2",
  "requestInfo": {
    "ipAddress": "204.16.5.19",
    "path": "/activate",
    "user": "TheDoctor"
  }
}
```

Which is nice but not ideal... having the ec2 instance ID and other fields was useful.

**Describe the solution you'd like**
A JSON Parser that merges all of the parsed fields with the original fields. 

**Describe alternatives you've considered**
I wish I could switch the order of my parsing filter and my record modifier filter. (The ec2 instance ID and etc come from a record modifier). It seems that these filters are applied in order by Fluent Bit. See the config files below. Unfortunately, for reasons which would take a while to explain, I can not switch the ordering of these filters. 

**Additional context**

Here are my Fluent Bit configuration files (slightly simplified to highlight my use case):

##### Main Config

```
[INPUT]
    Name forward
    unix_path /var/run/fluent.sock

[FILTER]
    Name record_modifier
    Match *
    Record ec2_instance_id i-079e527b553a30509
    Record ecs_cluster aggregator-public-subnets
    Record ecs_task_arn arn:aws:ecs:us-east-2:144718711470:task/32ce7459-bbac-4f94-b66f-6a72a6a6ff2c
    Record ecs_task_definition firelens-example-json:3

@INCLUDE /fluent-bit/etc/external.conf

[OUTPUT]
    Name firehose
    Match app-firelens*
    delivery_stream fluentd-service1
    region us-west-2
```

##### external.conf

```
[SERVICE]
    Log_Level debug
    Parsers_File /fluent-bit/parsers/parsers.conf

[FILTER]
    Name parser
    Match *
    Key_Name log
    Parser json
```

**Comment by edsiper** [comment]
@PettitWesley 

would you please give a try to the option ```Reserve_Data True``` in your filter parser ?

ref: https://docs.fluentbit.io/manual/filter/parser

**Comment by PettitWesley** [comment]
@edsiper Thanks!
